service/controller/v14/resource/endpoint/delete.go

```diff
@@ -37,23 +40,18 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 }
 
 func (r *Resource) NewDeletePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*controller.Patch, error) {
-	deleteChange, err := r.newDeleteChangeForDeletePatch(ctx, obj, currentState, desiredState)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
-	updateChange, err := r.newDeleteChangeForUpdatePatch(ctx, obj, currentState, desiredState)
+	deleteChange, err := r.newDeleteChange(ctx, obj, currentState, desiredState)
```

Comment:
This was super confusing. AFAIK one method did something and the other nothing. The only difference was one condition which caused the one method to do nothing at all. So I removed one and tried to get this more straight. We actually only want to delete in case there are no IPs within the endpoint anymore. No reason to make it more complicated.